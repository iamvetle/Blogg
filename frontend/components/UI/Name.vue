<template>
    <span data-test="name">
        {{ first_name }} {{ last_name }}        
    </span>
</template>

<script setup lang="ts">

defineProps<{
    first_name:string;
    last_name?:string;
}>();

</script>

<style scoped></style>