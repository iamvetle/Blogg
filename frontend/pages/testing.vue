<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>