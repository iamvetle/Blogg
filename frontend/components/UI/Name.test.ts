import Name from './Name.vue';
import { VueWrapper, shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

let wrapper: any;
let pinia: any = createTestingPinia();

let firstNameProp = "test_first_name"
let lastNameProp = "test_last_name"

const factory = () => {
    return shallowMount(Name, {
        global: {
            plugins: [pinia],
            components: {},
            mocks: {},
            stubs: {},
        },
        props: {
            first_name: firstNameProp,
            last_name: lastNameProp
        },
        slots: {}
    })
};

describe('Testing the name component', () => {

    beforeEach(() => {


    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
    });

    test('Should exist', () => {
        wrapper = factory()
        expect(wrapper.exists()).toBe(true)
    })
    test('Should have two name props', () => {
        wrapper = factory()

        expect(wrapper.props("first_name")).toBeTruthy()
        expect(wrapper.props("last_name")).toBeTruthy()
    })
    test('The first name prop should be correct ', () => {
        wrapper = factory()

        expect(wrapper.props("first_name")).toBe(firstNameProp)
    })
    test('The last name prop should be correct ', () => {
        wrapper = factory()

        expect(wrapper.props("last_name")).toBe(lastNameProp)
    })
    test('Should render the props', () => {
        wrapper = factory()

        const firstProp = wrapper.props("first_name")
        const lastProp = wrapper.props("last_name")
        
        expect(wrapper.text()).toContain(firstProp)
        expect(wrapper.text()).toContain(lastProp)
    })
    test('Should render the entire name', () => {
        wrapper = factory()

        const firstProp = wrapper.props("first_name")
        const lastProp = wrapper.props("last_name")

        const targetElement = wrapper.find("[data-test='name']")
        expect(targetElement.text()).toBe(`${firstProp} ${lastProp}`)
    })
    test('Should match snapshot', () => {
      wrapper = factory()

      expect(wrapper).toMatchSnapshot()
    })
});