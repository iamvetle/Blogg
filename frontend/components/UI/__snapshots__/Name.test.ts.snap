// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the name component > Should match snapshot 1`] = `"<span data-test=\\"name\\">test_first_name test_last_name</span>"`;
